tsdb/series_set.go

```diff
@@ -197,13 +197,33 @@ func (s *SeriesIDSet) UnmarshalBinary(data []byte) error {
 	return s.bitmap.UnmarshalBinary(data)
 }
 
+// UnmarshalBinaryUnsafe unmarshals data into the set.
+// References to the underlying data are used so data should not be reused by caller.
+func (s *SeriesIDSet) UnmarshalBinaryUnsafe(data []byte) error {
+	s.RLock()
+	defer s.RUnlock()
```

Comment:
Should this be a write lock?

Reply:
@zeebo I think we only need to protect access to bitmap, and not worry about mutations happening inside the roaring library.

Reply:
Ok. I asked because UnmarshalBinary above grabbed the write lock.

Reply:
Oh I see what you mean now. I guess this should be a lock, though it shouldn’t ever be called concurrently with another method on the type.

Reply:
Fixed in cb828f018. 👍